### PR TITLE
Add rename command

### PR DESCRIPTION
**Pull request by williamjamir**

### CODE REVIEW

File: module_renamer/commands/rename_imports.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    Main loop that interacts over all python files from the project and delegate
+    to an executor to parse each file
+
+    :param str project_path:
+        Path to the project that is going to be parsed.
+
+    :param str path_to_moved_imports_file:
+        Path of the file with the list of changed imports.
+    """
+    list_of_py_files = [i for i in walk_on_py_files(project_path)]
```
Comment (nicoddemus) [inline module_renamer/commands/rename_imports.py]: No need to change, but you can simply write: `list(walk_on_py_files(project_path))` as well 😉

File: module_renamer/commands/rename_imports.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            executor.submit(rename_file, py_file, path_to_moved_imports_file): py_file
+            for py_file in list_of_py_files
+        }
+
+        done_list = tqdm(futures.as_completed(future_map), total=file_counter, unit="files", leave=False)
+        for future in done_list:
+            try:
+                future.result()
+            except Exception as exc:
+                click.ClickException(exc)
```
Comment (nicoddemus) [inline module_renamer/commands/rename_imports.py]: Probably a better way is to create a list of errors and display them at the end as a summary of problems. Also, `click.ClickException(exc)` doesn't do anything, it only instantiates a `ClickException` and throws the result away, you probably meant `raise click.ClickException`. Also [`ClickException` accepts a `message` parameter](http://click.pocoo.org/5/api/#click.ClickException), not an exception object.

More info: http://click.pocoo.org/5/exceptions/

File: module_renamer/commands/rename_imports.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    :param str file_path:
+        Path of the file being parsed.
+    :param str path_to_moved_imports_file:
+        Path of the file with the list of changed imports.
+    """
+    list_with_moved_imports = _get_list_of_moved_imports(path_to_moved_imports_file)
+
+    with open(file_path, mode='r') as file:
+        tree = pasta.parse(file.read())
+        for moved_import in list_with_moved_imports:
```
Comment (nicoddemus) [inline module_renamer/commands/rename_imports.py]: `for old_path, new_path in list_with_moved_imports:` ;)

File: module_renamer/commands/rename_imports.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            old_path = moved_import[0]
+            new_path = moved_import[1]
+            try:
+                rename.rename_external(tree, old_path, new_path)
+            except ValueError:
+                click.ClickException("Some error happened on the following path: {0}.\n "
```
Comment (nicoddemus) [inline module_renamer/commands/rename_imports.py]: Again you probably forgot a `raise` here

**Review by nicoddemus — CHANGES_REQUESTED** [review]
We also need some docs, but I guess you plan to tackle this later right?

File: module_renamer/commands/rename_imports.py (hunk 3 above)
Comment (williamjamir) [reply module_renamer/commands/rename_imports.py]: Good catch =)

File: module_renamer/commands/rename_imports.py (hunk 2 above)
Comment (williamjamir) [reply module_renamer/commands/rename_imports.py]: - [x]  1) Create a list of errors and display them
- [x]  2) Raise the exception... :grin:

1) The list of errors is going to be displayed as below
```
Error: The following exception(s) has occurred while parsing the files:                                                                                                          

File /tmp/tests/file_a (copy).py generated an exception: invalid syntax (<unknown>, line 1)
File /tmp/tests/file_a.py generated an exception: invalid syntax (<unknown>, line 1)
File /tmp/tests/file_a (another copy).py generated an exception: invalid syntax (<unknown>, line 1)
```

2) I added a new test to check if the exception is being thrown ;)

File: module_renamer/commands/rename_imports.py (hunk 4 above)
Comment (williamjamir) [reply module_renamer/commands/rename_imports.py]: - [X] Fixed

File: module_renamer/commands/rename_imports.py (hunk 1 above)
Comment (williamjamir) [reply module_renamer/commands/rename_imports.py]: Thanks for the tip ;) 
I wasn't aware that list() could consume an iterable

**Review by nicoddemus — APPROVED** [review]

**Pull request merged**